gammapy/datasets/core.py

```diff
@@ -63,6 +63,10 @@ def mask(self):
 
     def stat_sum(self):
         """Total statistic given the current model parameters."""
+
+        if self.counts_off is None and not np.any(self.mask_safe.data) is False:
```

Comment:
In pratice, this is causing because `counts_off` does not belong to all the `Dataset` subclasses. This change has to be applied on the `MapdatasetOnOff` instead.

Reply:
Okay, thanks. I will adapt and push